### ISSUE TITLE
Performance issue: 7% load

### ISSUE THREAD

**Issue by T3rm1** [TheGuardianWolf/ThunderbirdTray#18]
As you can see in the image the program has a load of 7%. This is way too much.
![image](https://user-images.githubusercontent.com/5060247/79386042-d4667700-7f69-11ea-978d-b1634d422dd6.png)

**Comment by TheGuardianWolf** [comment]
Hey, I need some additional info to start looking at this, namely OS version, program version and framework and the state of the application. Does Task Manager say the same about the process? I personally have not found any performance issues, and there shouldn't be any since I'm using mostly .NET events

**Comment by T3rm1** [comment]
It doesn't happen all the time. I have not found a way to reliably reproduce this issue.
OS: Windows 7
.NET: 4.8.3761.0
Program version: 1.4.0.0
Thunderbird was hidden but it showed the same load when it was visible.

**Comment by TheGuardianWolf** [comment]
Honestly in that case it might be more of a .NET issue. After the initial polling and hooking process, everything is managed by the framework. If you ever encounter it again, have a look at the .NET performance and threads tabs, I'm not sure what's in there but it might be useful.

**Comment by sadluck** [comment]
Same issue here. Tried on two systems - Win10 64bit, Win7 64bit (very different computers). It doesn't happen right away after startup, but very often when I enter task manager much later with most of the system idling, ThunderbirdTray.exe is there sitting right on top with a CPU usage of about 7%.

**Comment by TheGuardianWolf** [comment]
As I suspected most of the usage is coming from within the .NET Framework. I don't think this is a fixable issue for me, I'd have to change the core of the stuff I am using.

![2020-04-25](https://user-images.githubusercontent.com/8168401/80267982-16c14e00-86f8-11ea-97f0-34be70f30150.png)

I'll have a look around for alternatives to UI Automation, but there might not be one.

**Comment by T3rm1** [comment]
So where you able to reproduce this issue or is this just profiling when the program behaves normally?
It looks like there are a lot of exceptions thrown in UIAutomation.dll. Maybe check what this is about.

**Comment by TheGuardianWolf** [comment]
I've already checked that, there doesn't seem to be anything I can do about it (https://stackoverflow.com/questions/6362367/wpf-ui-automation-issue). And no, I wasn't able to reproduce this issue, this is just it running normally.

To try to debug this, I'm sure I need to be running profiling on an affected machine, although I can guess that it is UIAutomation that is causing the issue.

I don't think I'm going to get further by trying to debug this myself, so I will need someone to help run the VS profiler while it is going on.

The best course of action is to look for a replacement for this rather obscure library, I'm in favour of going more into the Win32 API.

**Comment by TheGuardianWolf** [comment]
I released 1.6.0 with a polling workaround for minimise detection, could you see how that performs @T3rm1 @sadluck ?

**Comment by T3rm1** [comment]
No issues so far.

**Comment by vvld** [comment]
Hi, 
I just noticed the same thing. 
The problem occurs only after at least a few hours in which it is left in systray.
I am now trying the version 1.6.0 and I will come back with details (after a few days, to be sure). 
Thanks.

**Comment by vvld** [comment]
I left the computer on for 3 days and I can say the following:

Thunderbird x64 v68.7.0 (minimized in tray with "Minimize on Close" add-on)
ThunderbirdTray NetFramework 1.6.0
Windows 7 x64 (i5 CPU)
NetFramework 4.8.3761

Day 1 (UI Automation): After about 12h left in systray (all good), I started copying some files and that's it... ThunderbirdTray.exe - high cpu usage. 
From now on, regardless if thunderbird is minimized or visible, the cpu will fluctuate between 15-19% (see taskmanager screen - cpu-mem-I/O). I restarted it and tried to reproduce the behavior, but I didn't succeed. Exactly the same behavior as before with the 1.3.4 version of ThunderbirdTray NetFramework. Appears randomly (no matter if cpu is idle or not) after several hours or just after one or two.

Day 2 (Polling): All good. Cpu 00
Day 3 (Polling): All good. Cpu 00

Next month I will switch to windows 10. I hope everything is fine there too.
OK. That's about it. Thanks for app and I wish you all the best.

![tm](https://user-images.githubusercontent.com/61663995/81351178-6b87a000-90cc-11ea-873b-1341ce264faa.png)
